src/ruby/lib/grpc/core/call.rb

```diff
@@ -0,0 +1,20 @@
+module GRPC
+  module Core
+    class Call
+      # call-seq:
+      # metadata = call.metadata
+      #
+      # Gets the metadata object saved the call.
+      attr_reader :metadata
+
+      # call-seq:
+      # call.metadata = metadata
+      #
+      # Saves the metadata hash on the call.
+      def metadata=(val)
+        raise TypeError, "expected Hash, got #{val.class}" if val && !val.is_a?(::Hash)
```

Comment:
@chad I can see that, but `nil.class == NilClass`, so checking that `val.is_a?(Hash)` will in a way check that is not `nil`.

Anyway, I did a quick benchmark and `if val && !val.is_a?(Hash)` is 1.32 times faster than `unless val.is_a?(Hash)`, at least in my machine.